### PR TITLE
Refactored Hooks API

### PR DESCRIPTION
**Pull request by nicosomb**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 9.0.x 
| Description?      | Refactored Hooks API: delete hook-status endpoint and change /hooks/id to /hook/id
| Type?             | improvement 
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI is green
| UI Tests          | https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/12419408328
| Fixed issue or discussion?     | N/A
| Related PRs       | https://github.com/PrestaShop/ps_apiresources/pull/43
| Sponsor company   | PrestaShop SA

### CODE REVIEW

File: tests/UI/campaigns/functional/API/01_basicTest.ts [hunk 1]
```diff
@@ -23,12 +23,5 @@ describe('API : Basic Test', async () => {
       const apiResponse = await apiContext.get('');
       expect(apiResponse.status()).to.eq(404);
     });
-
-    it('should request the endpoint /hook-status', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/API/01_basicTest.ts:27]: Please replace hook-status by hooks

(We don't test the endpoint but the access)

File: tests/UI/campaigns/functional/API/01_clientCredentialGrantFlow/01_internalAuthServer/02_resourceEndpoint.ts [hunk 2]
```diff
@@ -122,62 +119,6 @@ describe('API : Internal Auth Server - Resource Endpoint', async () => {
...
       expect(jsonResponse.access_token).to.be.a('string');
-
-      accessToken = jsonResponse.access_token;
-      accessTokenExpired = utilsAPI.setAccessTokenAsExpired(accessToken);
-    });
-
-    it('should request the endpoint /hook-status/1 without access token', async function () {
-      await testContext.addContextItem(this, 'testIdentifier', 'requestEndpointWithoutAccessToken', baseContext);
-
-      const apiResponse = await apiContext.get('hook-status/1');
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/API/01_clientCredentialGrantFlow/01_internalAuthServer/02_resourceEndpoint.ts:133]: Please replace hook-status by hooks

(We don't test the endpoint but the access)

**Review by Progi1984 — CHANGES_REQUESTED** [review]
Please don't remove UI Tests but change the called endpoint

**Review by Hlavtox — DISMISSED** [review]
You cannot do this, this command and handler is used to control hook status in backoffice > module positions page.

**Review by Progi1984 — DISMISSED** [review]
Approved for UI Tests

**Review by boherm — DISMISSED** [review]

File: tests/UI/campaigns/functional/API/01_basicTest.ts [hunk 3]
```diff
@@ -24,10 +24,10 @@ describe('API : Basic Test', async () => {
       expect(apiResponse.status()).to.eq(404);
     });
 
-    it('should request the endpoint /hook-status', async function () {
+    it('should request the endpoint /hook', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/API/01_basicTest.ts]:
```suggestion
    it('should request the endpoint /hooks', async function () {
```

File: tests/UI/campaigns/functional/API/01_basicTest.ts [hunk 4]
```diff
@@ -24,10 +24,10 @@ describe('API : Basic Test', async () => {
...
       await testContext.addContextItem(this, 'testIdentifier', 'requestNewApiHookStatus', baseContext);
 
-      const apiResponse = await apiContext.get('hook-status');
+      const apiResponse = await apiContext.get('hook');
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/API/01_basicTest.ts]:
```suggestion
      const apiResponse = await apiContext.get('hooks');
```

File: tests/UI/campaigns/functional/API/01_basicTest.ts [hunk 5]
```diff
@@ -24,10 +24,10 @@ describe('API : Basic Test', async () => {
       expect(apiResponse.status()).to.eq(404);
     });
 
-    it('should request the endpoint /hook-status', async function () {
+    it('should request the endpoint /hooks', async function () {
       await testContext.addContextItem(this, 'testIdentifier', 'requestNewApiHookStatus', baseContext);
 
-      const apiResponse = await apiContext.get('hook-status');
+      const apiResponse = await apiContext.get('hooks');
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/API/01_basicTest.ts]:
```suggestion
      const apiResponse = await apiContext.put('hooks');
```

File: tests/UI/campaigns/functional/API/01_basicTest.ts [hunk 6]
```diff
@@ -24,10 +24,10 @@ describe('API : Basic Test', async () => {
       expect(apiResponse.status()).to.eq(404);
     });
 
-    it('should request the endpoint /hook-status', async function () {
+    it('should request the endpoint /hooks', async function () {
       await testContext.addContextItem(this, 'testIdentifier', 'requestNewApiHookStatus', baseContext);
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/API/01_basicTest.ts]:
```suggestion
      await testContext.addContextItem(this, 'testIdentifier', 'requestMethodNotAllowed', baseContext);
```

**Comment by nicosomb** [comment]
@Progi1984 🫶

**Review by boherm — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Pull request merged**